internal/jimm/jimm.go

```diff
@@ -236,6 +237,9 @@ type JujuManager interface {
 	UpdateMigratedModel(ctx context.Context, user *openfga.User, modelTag names.ModelTag, targetControllerName string) error
 	ValidateModelUpgrade(ctx context.Context, u *openfga.User, mt names.ModelTag, force bool) error
 
+	// Migration related methods
+	Prechecks(ctx context.Context, user *openfga.User, model migration.ModelInfo) error
```

Comment:
Could you move preparemodelmigration above this? And can we keep the order in which they run here with a little godoc explaining it? I think it'd help us for future aha.